lib/wibox/widget/calendar.lua

```diff
@@ -331,6 +362,7 @@ local function get_calendar(type, date, font)
     ret._private.week_numbers  = beautiful.calendar_week_numbers or false
     ret._private.start_sunday  = beautiful.calendar_start_sunday or false
     ret._private.long_weekdays = beautiful.calendar_long_weekdays or false
+    ret._private.flex_height   = beautiful.calendar_flex_height or true
```

Comment:
Shouldn't this be?

```suggestion
    ret._private.flex_height   = beautiful.calendar_flex_height or false
```

It's documented as `[opt=false]`.

Reply:
Hum, that's embarrassing. What I meant was: 

```suggestion
    flex_height                = beautiful.calendar_flex_height
    ret._private.flex_height   = if flex_height ~= nil then flex_height else true
```
(And change the doc line too)

If we set that to false, we change the default behavior of the calendar. Maybe this is what we want? In that case we can default to false.

Reply:
It's true that, technically, the cells always had a "flex" height and that this PR didn't change that. But from a user's POV, that _is_ what would be changing currently.

If you look at [your second comment](https://github.com/awesomeWM/awesome/pull/3387#issuecomment-874917447), the change in code might be "remove empty lines", but the change in behaviour is "stretch cells for short months". And as I already mentioned, adding empty lines in a year view seems to be the norm.

Additionally, the initial bug about extraneous lines you referenced is only about the month view, where there really is no need for those empty lines. Because of that, I would consider a default of `true` a change in behaviour outside the scope of the issue that's being fixed.

Reply:
Good points, then false it is